### PR TITLE
Remove items from trakt lists upon successful download/import

### PR DESCRIPTION
**Pull request by geogolem**
This PR is being opened for discussion purposes -- its likely not ready to be merged although I believe the current implementation to be fully functional there might be some improvements.
#### Database Migration
YES

#### Description
Please see the discussion here: https://github.com/Radarr/Radarr/issues/1362

#### Todos
- [ ] Tests

#### Issues Fixed or Closed by this PR
https://github.com/Radarr/Radarr/issues/1362
* #

### CODE REVIEW

**Comment by jkaberg** [comment]
@geogolem Would it be possible to mark the release as collected @ trakt aswell? Eg. remove wishlist "tag" and add "collected"?

File: src/NzbDrone.Core/NetImport/Trakt/TraktRequestGenerator.cs [hunk 1]
```diff
@@ -30,6 +35,65 @@ public TraktRequestGenerator()
...
+                    var ids = new Ids();
+
+                    ids.imdb = movie.ImdbId;
+                    mov.ids = ids;
+                    postData.movies = moviesList;
+                    postData.movies.Add(mov);
+
+
+                    listRemovalRequest.SetContent(Json.ToJson(postData));
+                    HttpClient.Execute(listRemovalRequest);
```
Comment (galli-leo) [inline src/NzbDrone.Core/NetImport/Trakt/TraktRequestGenerator.cs]: This is a request generator class. We should only return the request not execute it here.

File: src/NzbDrone.Core/NetImport/RSSImport/RSSImport.cs [hunk 2]
```diff
@@ -12,6 +12,7 @@ public class RSSImport : HttpNetImportBase<RSSImportSettings>
         public override string Name => "RSSList";
         public override bool Enabled => true;
         public override bool EnableAuto => false;
+		public override bool ShouldClean => false;
```
Comment (galli-leo) [inline src/NzbDrone.Core/NetImport/RSSImport/RSSImport.cs]: We should have a property similar to indexers called CanClean. This way the switch can be disabled in the ui, as to not confuse users why their tmdb list is not getting cleaned.

File: src/UI/Settings/NetImport/Edit/NetImportEditViewTemplate.hbs [hunk 3]
```diff
@@ -60,6 +60,29 @@
 										</div>
 								</div>
 						</div>
+
+                        <div class="form-group">
+                                <label class="col-sm-3 control-label">Remove Movies from this List after Successful Download/Import</label>
+
+                                <div class="col-sm-5">
+                                        <div class="input-group">
+                                                <label class="checkbox toggle well">
```
Comment (galli-leo) [inline src/UI/Settings/NetImport/Edit/NetImportEditViewTemplate.hbs]: We should disable this button, if the clean method is not implemented.

**Comment by jkaberg** [comment]
@geogolem any news on this PR?

**Comment by geogolem** [comment]
https://github.com/Radarr/Radarr/issues/2469

**Pull request closed without merging**